Part of #18714 and #17670: Changes to nfts-detection-notice.js

dhruvv173 opened the pull request:
## Explanation
This PR is a part of #18714 and #17670 
Replaced Typography with Text component and deprecated Typography consts with enums for the file: ```nfts-detection-notice.js```

- ui/components/app/nfts-detection-notice/nfts-detection-notice.js

## Screenshots/Screencaps

<!-- If you're making a change to the UI, make sure to capture a screenshot or a short video showing off your work! -->

### Before

<!-- How did the UI you changed look before your changes? Drag your file(s) below this line: -->
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/ad7ba075-c3b9-4813-a0ea-faa199dacf86)

### After

<!-- How does it look now? Drag your file(s) below this line: -->
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/8b195152-f0c7-4bfd-9f2d-b80f5e42a46f)

## Manual Testing Steps

<!--
How should reviewers and QA manually test your changes? For instance:

- Go to this screen
- Do this
- Then do this
-->

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

dhruvv173 committed:
Changes to nfts-detection-notice.js

dhruvv173 commented:
Hey @georgewrmarshall
Please review this PR. The ```nfts-detection-notice.js``` does not exist on Storybook so I couldnt attach the before/after screenshots.
Thanks!

georgewrmarshall reviewed (commented):
Hey @dhruvv173, code is looking good but could you please add some before/ after screenshots to ensure that there are no visual regressions as outlined in the acceptance criteria for the issue. Thanks!

georgewrmarshall commented:
> Hey @georgewrmarshall Please review this PR. The `nfts-detection-notice.js` does not exist on Storybook so I couldnt attach the before/after screenshots. Thanks!
> 
> Edit: 2 of my PRs today are failing are failing Label PR test due to (Resource not accessible by integration) error, otherwise everything works fine.

You can create a story if you like. Documentation here: https://metamask.github.io/metamask-storybook/?path=/story/getting-started-documentation-guidelines--page#creating-a-story

No need for arg types or MDX docs

```
import React from 'react';
import NftsDetectionNotice from '.';

export default {
  title: 'Components/App/NftsDetectionNotice',
};

export const DefaultStory = () => <NftsDetectionNotice />;

DefaultStory.storyName = 'Default';
```

georgewrmarshall committed:
Adding story

georgewrmarshall reviewed (approved):
Hey @dhruvv173, I've added a story so we can do a visual check. It LGTM! Thanks for your contribution

<img width="1440" alt="Screenshot 2023-05-09 at 5 03 44 PM" src="https://user-images.githubusercontent.com/8112138/237020094-b732e8c2-d8ba-43ef-b98b-0ff16abcc7ea.png">

georgewrmarshall commented:
I think what you've done here is add 2 screenshots of the "After" changes. What you would need to do is create the story on the develop branch and take a screenshot there.

dhruvv173 commented:
> I think what you've done here is add 2 screenshots of the "After" changes. What you would need to do is create the story on the develop branch and take a screenshot there. No need to update this PR the code is good to go.

Should I create a new fork, add the code for story there, run ```yarn storybook``` and then add the screenshots?
Sorry for the inconvenience caused :/

georgewrmarshall commented:
@dhruvv173 yes that sounds right. It should be off of the `develop` branch

dhruvv173 commented:
Hey @georgewrmarshall 
I was working on file named `gas-timing.component`, replaced Typography with Text, added the new enums and created a story as it does not exist already but it does not load anything (refer the image below)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/bc5945a3-a88d-476c-af67-287228cea9e8)
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/32e03b06-95cb-4d48-bb4a-0df8a2e38e3e)

added the story file below
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/37dc1775-e729-4fe8-b1f1-335303c2816a)

or is it because the snapshot of `gas-timing.component.test.js.snap` has just one `<div />` tag ?
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/656e3e8f-4b77-449e-bc99-16fb9cc7560e)

I also tried updating the screenshot with `yarn jest ui/components/app/gas-timing --updateSnapshot` command but that didnt work either.
Could you please check this once?
Thank you!

georgewrmarshall commented:
Hey @dhruvv173, you have to pass the component the write data either through props or mock redux state using the decorator and provider.

dhruvv173 commented:
> Hey @dhruvv173, you have to pass the component the write data either through props or mock redux state using the decorator and provider.

Okay I'll do that and submit a PR.
Could you please check this [#19115](https://github.com/MetaMask/metamask-extension/pull/19115) once when possible?
Thanks!
